task/control-plane-agent/src/mgs_common.rs

```diff
@@ -97,6 +101,12 @@ impl MgsCommon {
         Ok(state)
     }
 
+    /// The SP, having reset itself, will not be able to respond to
+    /// the later reset_trigger message.
+    ///
+    /// So, after getting an ACK for the prepare message, MGS will send and
+    /// retry the reset_trigger message until it gets rejected for lack of
+    /// a corresponding prepare message.
```

Comment:
I'd suggest modifying these functions to call into `reset_component_prepare` and `reset_component_trigger`, deleting `self.reset_requested` (since `self.reset_component_requested` serves the same purpose).

Reply:
Good suggestion. I did you one better though! After chatting with @jgallagher this morning, I updated MGS to not call these functions anymore and instead use the component versions like this: https://github.com/oxidecomputer/management-gateway-service/commit/cf6bf75986289892eccd562231afea197539a30b#diff-4e2f0f9389ed68d983518b63098811d189767e0b271891d22cf29aa2a77794c4R783-R788

This means I was able to remove `reset_component` and `reset_trigger` altogether in https://github.com/oxidecomputer/hubris/pull/1261/commits/127cbea286c342efe0af81eed651411bc4e33d95